src/components/Avatar/Avatar.tsx

```diff
@@ -25,12 +25,43 @@ type AvatarProps = (
   | JSX.IntrinsicElements["button"]
   | JSX.IntrinsicElements["span"]
 ) & {
+  /**
+   * The avatar image URL
+   */
   src?: React.ComponentProps<typeof SuspenseImg>["src"];
+  /**
+   * The Matrix ID, Room ID, or Alias to generate the color when no image source
+   * is provided. Also used as a fallback when name is empty
+   */
   id: string;
+  /**
+   * The name used for the initial letter displayed when no image source is provided
+   */
   name: string;
+  /**
+   * Defines the avatar type, typically round, square is usually for spaces
+   * @default "round"
+   */
   type?: "square" | "round";
+  /**
+   * The avatar size
+   */
   size?: CSSStyleDeclaration["height"];
+  /**
+   * On click handler, will turn the avatar into a button
+   */
   onClick?: (e: React.MouseEvent) => void;
+  /**
+   * Key down handler, will turn the avatar into a button
+   */
+  onKeyDown?: (e: React.KeyboardEvent) => void;
+  /**
+   * Key up handler, will turn the avatar into a button
+   */
+  onKeyUp?: (e: React.KeyboardEvent) => void;
```

Comment:
Can these not be inferred from the button & span props?

Reply:
They can, they are only here for the JSDoc.